### PR TITLE
Fix Psycopg2 pool stubs

### PR DESCRIPTION
**Pull request by ewjoachim**
Documentation: https://www.psycopg.org/docs/pool.html
Issue encountered: 
```
procrastinate/psycopg2_connector.py:188: error: "AbstractConnectionPool" has no attribute "getconn"
procrastinate/psycopg2_connector.py:195: error: "AbstractConnectionPool" has no attribute "putconn"
procrastinate/psycopg2_connector.py:167: error: "AbstractConnectionPool" has no attribute "closeall"
```

Disclaimer: I have spent very limited amount of time on this PR, and if there's a need to change anything, I'll be glad to spend some more time on it. I'm not sure how to test the change itself.

Ping @JukkaL who created this file.

### CODE REVIEW

**Comment by srittau** [comment]
Thanks. The file needs to be reformatted with black and the abstract methods need to be added to `stubs/psycopg2/@tests/stubtest_allowlist.txt`, because they don't exist at runtime. (See the CI failures.) We should also add a comment that the methods are abstract and don't exist at runtime in the base class.

**Comment by JelleZijlstra** [comment]
Formatting issues now get fixed automatically. We still need to make the other changes @srittau mentioned though.

**Comment by ewjoachim** [comment]
> Formatting issues now get fixed automatically

That's awesome. Long live pre-commit.ci.

I'll take care of the other comments :)

File: stubs/psycopg2/psycopg2/pool.pyi [hunk 1]
```diff
@@ -9,14 +9,11 @@ class AbstractConnectionPool:
...
-    putconn: Any
-    closeall: Any
-
-class ThreadedConnectionPool(AbstractConnectionPool):
-    def __init__(self, minconn, maxconn, *args, **kwargs) -> None: ...
+    # getconn, putconn and closeall are officially documented as methods of the
+    # abstract base class, but in reality, they only exist on the children classes
     def getconn(self, key: Any | None = ...): ...
-    def putconn(self, conn: Any | None = ..., key: Any | None = ..., close: bool = ...) -> None: ...
+    def putconn(self, conn, key: Any | None = ..., close: bool = ...) -> None: ...
```
Comment (ewjoachim) [inline stubs/psycopg2/psycopg2/pool.pyi:15]: CI requested that I removed the default value for the `conn` argument. This is in line with the documentation so I applied it.

**Comment by ewjoachim** [comment]
Haha. If I put a default None to putconn, I get:
`error: psycopg2.pool.SimpleConnectionPool.putconn is inconsistent, stub argument "conn" has a default value but runtime argument does not`
If I don't I get:
`error: psycopg2.pool.ThreadedConnectionPool.putconn is inconsistent, runtime argument "conn" has a default value but stub argument does not`

:rofl: 
I'm going to specify the methods differently for the class that doesn't resepct the doc.

**Comment by ewjoachim** [comment]
I'll take my builds in any color as long as it's green :D

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**